### PR TITLE
ENB: split examples from doc examples

### LINKED ISSUE #1142: ENB: split examples from inline examples
Now we build all examples using `magic run examples` command.

After build will be created `desktop.examples` and `touch.examples` directories. They will be as user examples and examples are only needed for documentation.

My proposal:
1. Build user examples (`block.examples/example-name/example-name.bemjson.js`) using `examples` task and write result to `platform.examples` dir.
2. Build examples for docs using `doc-examples` task and write result to `platform.docs/examples` dir.
3. Build docs + doc examples using `docs` task.

### PR DESCRIPTION
**Pull request by blond**
Resolved #1142

### CODE REVIEW

File: .enb/tasks/docs.js [hunk 1]
```diff
@@ -1,45 +1,38 @@
-var config = require('../config'),
-    PLATFORMS = config.platforms;
+var path = require('path'),
+    vow = require('vow');
 
 /**
- * Creates `docs` task.
+ * Creates meta task `docs`.
```
Comment (tadatuta) [inline .enb/tasks/docs.js]: Creates `docs` meta task

File: .enb/tasks/docs.js [hunk 2]
```diff
@@ -1,45 +1,38 @@
...
  *
- * This task allows you to build docs.
+ * This task allows you to build docs with examples.
```
Comment (tadatuta) [inline .enb/tasks/docs.js]: ~~you~~

File: .enb/tasks/docs.js [hunk 3]
```diff
@@ -1,45 +1,38 @@
...
 module.exports = function (project) {
-    // load plugin
-    project.includeConfig('enb-bem-docs');
-    var plugin = project.module('enb-bem-docs'),
-        // create task with `docs` name
-        // and get helper to configure it
-        helper = plugin.createConfigurator('docs'),
-        langs = project.getLanguages();
+    // load service task configs
+    ['__docs__', '__doc-examples__'].forEach(function (name) {
```
Comment (tadatuta) [inline .enb/tasks/docs.js:17]: won't it be simplier to just copy/paste:

``` js
project.includeConfig(path.join(__dirname, 'docs', '__docs__.js'));
project.includeConfig(path.join(__dirname, 'docs', '__doc-examples__.js'));
```

?

File: .enb/tasks/docs/__doc-examples__.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
+var path = require('path'),
+    naming = require('bem-naming'),
+    config = require('../../config'),
+    configurePage = require('../../helpers/page'),
+    PLATFORMS = config.platforms;
+
+/**
+ * Creates service `__doc-examples__` task.
+ *
+ * This task allows you to build examples from md files.
```
Comment (tadatuta) [inline .enb/tasks/docs/__doc-examples__.js]: ~~you~~

File: .enb/tasks/docs/__doc-examples__.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    }
+    var plugin = project.module('enb-bem-examples'),
+        // create task with `examples` name
+        // and get helper to configure it
+        helper = plugin.createConfigurator('__doc-examples__');
+
+    PLATFORMS.forEach(function (platform) {
+        var dirPattern = platform + '.doc-examples/*/*';
+
+        // configure of build BEMJSON files
```
Comment (tadatuta) [inline .enb/tasks/docs/__doc-examples__.js]: configure BEMJSON files building

File: .enb/tasks/docs/__doc-examples__.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        configure(helper, platform);
+
+        // configure of build pages by BEMJSON files
```
Comment (tadatuta) [inline .enb/tasks/docs/__doc-examples__.js]: configure pages building by BEMJSON files

File: .enb/tasks/docs/__doc-examples__.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        project.nodes(dirPattern, function (node) {
+            configurePage(node, {
+                platform : platform
+            });
+        });
+    });
+};
+
+/**
+ * Configures build BEMJSON files.
```
Comment (tadatuta) [inline .enb/tasks/docs/__doc-examples__.js]: Configures BEMJSON files building

File: .enb/tasks/docs/__docs__.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
+var config = require('../../config'),
+    PLATFORMS = config.platforms;
+
+/**
+ * Creates service `__docs__` task.
+ *
+ * This task allows you to build docs without examples.
```
Comment (tadatuta) [inline .enb/tasks/docs/__docs__.js]: ~~you~~

**Comment by blond** [comment]
:up:

**Pull request merged**